### ISSUE TITLE
[mwc-list-item] twoline wrongly indents text

### ISSUE THREAD

**Issue by yinonov** [material-components/material-web#2425]
**Describe the bug**
list item of two lines adds undesired space and causes indentation to text due to `::before` pseudo-element 

**To Reproduce**
1. Go to [list item with `twoline` demo](https://material-components.github.io/material-components-web-components/demos/list/)
2. See indented text in each list item

**Expected behavior**
text should be stick to start of line

**Screenshots**
![image](https://user-images.githubusercontent.com/15140652/118618187-092baa00-b7cc-11eb-89bf-a8c398654574.png)

**Comment by gullerya** [comment]
Probably a bit more info on this:
- In Sarari (14 still happens) it may cause an unnecessary ellipsis in the text, and see below TLDR

TLDR;

It looks like the additional spacing is added when (and due) to a white spaces around the text, supplied by the consumer.
Although consumer may take extra care and remove any white spacing / tabs / new-lines around, this is inconvenient due to HTML templating practices, prettifiers etc.

The added space shown above happens due to coexistence of 3:
- `list-item__primary / secondary-text` is `display: block`
- text content of `list-item__primary / secondary-text` has white space chars before the text
- `list-item__primary / secondary-text` is NOT a first child in the parent

Last point explains how this all is related to the usage of `:before` pseudo element, it just has a side effect of not trimming outer white spaces.

Those points also explain how this is NOT happening in one line scenario (no `:before`) or when the text is carefully set into the `textContent` by the consuming application (no leading white spaces).

**Comment by e111077** [comment]
should be handled in list migration

**Comment by yinonov** [comment]
is it planned for 0.22.0?

**Comment by e111077** [comment]
Unfortunately likely not. The next release we expect to simply be a release where we stamp a tag as the final Lit 1 release. Then 0.23.0 will be the Lit 2 release.

We are expecting to have the list migration done for sure by 1.0.0.

Though list > menu > select are some of the most-complex components in our libraries so we expect this will likely be further down the line.

**Comment by asyncLiz** [comment]
Should be fixed in M3
